src/lib.rs

```diff
@@ -5,6 +5,63 @@
 //! - The [`ProtoComponent`](components::ProtoComponent) trait provides methods to load components from assets.
 //! - The [`ProtoDeserializer`](data::ProtoDeserializer) trait describes component deserialization.
 //! - [`ProtoPlugin`](plugin::ProtoPlugin) provides configuration for asset loading.
+//!
+//! # Examples
+//! Define a serialized prototype:
```

Comment:
I'd prefer if these were broken down into multiple code snippets like in the README so it's easier to digest. Like:

Define a serialized prototype:
```yaml
// assets/prototypes/simple-enemy.yaml
...
```

Implement `ProtoComponent` ...
```rust
#[derive(Clone, Serialize, Deserialize, ProtoComponent, Component)]
struct Enemy;
...
```

etc.

Reply:
> it's better to just write out a manual example imo.
I tend to agree that the front page docs should be a little trimmer.